runtime/src/main/java/com/dylibso/chicory/runtime/wasi/WasiP1.java

```diff
@@ -0,0 +1,756 @@
+package com.dylibso.chicory.runtime.wasi;
+
+import static com.dylibso.chicory.wasm.types.ValueType.*;
+
+import com.dylibso.chicory.runtime.HostFunction;
+import com.dylibso.chicory.runtime.Memory;
+import com.dylibso.chicory.runtime.exceptions.WASMRuntimeException;
+import com.dylibso.chicory.wasm.types.Value;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.List;
+
+public class WasiP1 {
+    private static final System.Logger LOGGER = System.getLogger(WasiP1.class.getName());
+
+    private WasiOptions options;
+
+    public WasiP1() {
+        // TODO by default everything should by blocked
+        // this works now because streams are null.
+        // maybe we want a more explicit way of doing this though
+        this(new WasiOptions());
+    }
+
+    public WasiP1(WasiOptions opts) {
+        this.options = opts;
+        LOGGER.log(
+                System.Logger.Level.WARNING,
+                "Use of WASIP1 is experimental and will only work for the simplest of use cases.");
+    }
+
+    public HostFunction[] toHostFunctions() {
+        var fidx = 55;
+        var functions = new HostFunction[fidx + 1];
+
+        functions[fidx--] =
```

Comment:
I stubbed out as many WASI p1 functions as i could find. Oftentimes they are imported and never called. But if they are, it should trap here with the name of the wasi function that is not implemented.

The only real implementations at the moment are fd_read and fd_write and only for file descriptors 0,1, and 2